projects/ccf-rui/src/app/modules/content/video-modal/video-modal/video-modal.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, OnInit, HostBinding, Inject } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+
+/**
+ * Component for displaying a youtube video inside of an angular material modal.
+ */
+@Component({
+  selector: 'ccf-video-modal',
+  templateUrl: './video-modal.component.html',
+  styleUrls: ['./video-modal.component.scss']
+})
+export class VideoModalComponent implements OnInit {
+  /** HTML class name */
+  @HostBinding('class') readonly clsName = 'ccf-video-modal';
+
+  /**
+   * Creates an instance of video modal component.
+   */
+  constructor(
+    public dialogRef: MatDialogRef<VideoModalComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: unknown
+  ) { }
+
+  /**
+   * load the youtube player api in on init
+   */
+  ngOnInit(): void {
+    this.loadYoutubePlayerAPI();
+  }
+
+  /**
+   * Loads in youtube player api
+   */
+  loadYoutubePlayerAPI(): void {
```

Comment:
Yuck. Does the angular youtube API not hide this junk? If not, that's ok, but *yuck*. There might be an angular way to do this if not.

Reply:
I think it would be better to add the script tag in index.html. Add `async` and/or `defer` to the script tag to load asynchronously.

Reply:
I would rather not add to the index.html, because it'll complicate the embedding story.

Reply:
Use Renderer2 <https://angular.io/api/core/Renderer2> to make the script tag